internal/cmd/request.go

```diff
@@ -105,13 +107,107 @@ func PrintCMDResponseFull(request any, response core.APIResponse, responseType a
 		fmt.Printf("\nAutograder Response:\n---\n%s\n---\n", util.MustToJSONIndent(response))
 	}
 
+	successfulConversion := false
+	customPrintOutput := ""
 	if customPrintFunc != nil {
-		fmt.Println(customPrintFunc(response))
-	} else if responseType == nil {
-		fmt.Println(util.MustToJSONIndent(response.Content))
+		customPrintOutput, successfulConversion = customPrintFunc(response)
+	}
+
+	output := ""
+	if successfulConversion {
+		output = customPrintOutput
 	} else {
-		responseContent := reflect.New(reflect.TypeOf(responseType)).Interface()
-		util.MustJSONFromString(util.MustToJSON(response.Content), &responseContent)
-		fmt.Println(util.MustToJSONIndent(responseContent))
+		if responseType != nil {
+			responseContent := reflect.New(reflect.TypeOf(responseType)).Interface()
+			util.MustJSONFromString(util.MustToJSON(response.Content), &responseContent)
+			output = util.MustToJSONIndent(responseContent)
+		} else {
+			output = util.MustToJSONIndent(response.Content)
+		}
+	}
+
+	fmt.Println(output)
+}
+
+// Attempt to convert an API response into a TSV table.
+// If the response content has multiple keys or a single map, it creates a single row table.
+// If it has one key with a list, the values in each element becomes a row of the table.
+// Return the table and true on a successful conversion or false if there is an issue converting.
+func ConvertApiResponseToTable(response core.APIResponse) (string, bool) {
```

Comment:
s/Api/API/

In this project we usually do all caps for acronyms.
This is a super project-by-project (and sometimes file-by-file) thing.
I didn't even know what we did and just grepped for 'Api' vs 'API'.